core/src/stored/append.cc

```diff
@@ -372,7 +590,10 @@ bool DoAppendData(JobControlRecord* jcr, BareosSocket* bs, const char* what)
   /* Check if we can still write. This may not be the case
    * if we are at the end of the tape or we got a fatal I/O error. */
   if (ok || dev->CanWrite()) {
-    jcr->JobFiles = file_currently_processed.GetFileIndex();
+    // take into account the fact that GetFileIndex() may return -1
+    // if no file is currently getting processed.
+    // This should only happen if no file was send to begin with!
+    jcr->JobFiles = std::max(file_currently_processed.GetFileIndex(), 0);
```

Comment:
was there a bug you wanted/needed to fix here?

Reply:
Yes, but only a visual one.   If you start a job that does not back up any files (for some reason) then the joblog will say that the fd send 0 files but the sd wrote 4 billion of them.